cli/lib/kontena/callbacks/master/deploy/56_set_server_provider_after_deploy.rb

```diff
@@ -0,0 +1,27 @@
+module Kontena
+  module Callbacks
+    class SetServerProviderAfterDeploy < Kontena::Callback
+
+      include Kontena::Cli::Common
+
+      matches_commands 'master create'
+
+      def after
+        return unless command.exit_code == 0
+        return unless config.current_master
+        return unless config.current_master.name == command.result[:name]
+        return unless command.result[:provider]
+
+        require 'shellwords'
+
+        cmd = ['master', 'config', 'set', "server.provider=#{command.result[:provider]}"]
+        spinner "Setting Master configuration server.provider to '#{command.result[:provider]}'" do
+          Retriable.retriable do
+            Kontena.run(cmd.shelljoin)
```

Comment:
`Kontena.run` would run callbacks if there were any. Also instance variables inside the method would work. If you directly call a method on a command, then you need to write the method so that they don't use any instance variables (options/flags/params).


The ssh improvement PR #1205 makes it possible to pass an array instead of a string to `Kontena.run` so you don't have to use shellescape/shelljoin.